File: julia/main3.jl
function dangerous(array::Vector{Int}, index::Int)::Int
    return array[index]  # BoundsError
end

function foo(array::Vector{Int}, counter::Int)::Int
    if counter == 0
        return dangerous(array, counter + 9137)
    else
        return foo(array, counter - 1)
    end
end

a = fill(0, 1000)
r = foo(a, 900)
println("The result is ", r)

File: swift/main3.swift
func dangerous(_ array: [Int], _ index: Int) -> Int {
    return array[index] // traps on OOB
}

func foo(_ array: [Int], _ counter: Int) -> Int {
    if counter == 0 { return dangerous(array, counter + 9137) }
    return foo(array, counter - 1)
}

let array = Array(repeating: 0, count: 1000)
let r = foo(array, 900)
print("The result is \(r)")
